### DISCUSSION TITLE
Have modal not take up entire viewable screen

### DISCUSSION THREAD

**Discussion by philiplindberg** [react-navigation/react-navigation#10849] [Q&A]
I'm trying to present a modal in my web/mobile expo app using a `StackNavigator`. On iOS and Android, the fullscreen modal works great since there's limited screen real estate to work with, but on web I'd love the option to display the modal as more of a window in the middle of the screen with a semi-transparent backdrop behind it.

I've played around with a bunch of the `StackNavigator`'s `screenOptions` as documented [here](https://reactnavigation.org/docs/stack-navigator), but I just can't get the modal to display within a limited part of the screen. I was hoping the `cardStyleInterpolator` method would give me more flexibility when customizing the modal by tweaking the `containerStyle`, `cardStyle` and `overlayStyle` properties, but so far I haven't been able to get it to work.

Is this even possible with the built-in modal?

**Comment by satya164** [comment]
https://reactnavigation.org/docs/stack-navigator/#transparent-modals

**Reply by philiplindberg** [reply]
So I actually almost got it working by setting the screen presentation to `transparentModal` (thanks for pointing me in the right direction!) and tweaking some other settings. Here's an expo snack of my "modal dialog" in its current state:

https://snack.expo.dev/@philiplindberg/modal-dialog-demo

As you can see in the demo, I'm able to display the modal centered horizontally on the screen, but the thing I can't quite figure out is how to adjust its height so it doesn't take up the entire viewport's height. Any suggestions there?
